monitor/monitor.go

```diff
@@ -124,6 +168,9 @@ func (m *Monitor) Run(npages int, agentPipe io.Reader) {
 
 // dumpStat prints out the monitor status in JSON.
 func (m *Monitor) dumpStat() {
+	mutex.Lock()
```

Comment:
Good catch.

Reply:
Embarrassingly, I caught this on my 3rd redo of this PR :/